test/integration/policyConsumptionTest.ts

```diff
@@ -0,0 +1,149 @@
+import { strict as assert } from "assert";
+
+import { newTestUser } from "./clientHelper";
+import { getMessagesByUserIn } from "../../src/utils";
+import config from "../../src/config";
+import axios from "axios";
+import { LogService } from "matrix-bot-sdk";
+import { createBanList, getFirstReaction } from "./commands/commandUtils";
+
+/**
+ * Get a copy of the rules from the ruleserver.
+ */
+async function currentRules() {
+    return await (await axios.get(`http://${config.web.address}:${config.web.port}/api/1/ruleserver/updates/`)).data
+}
+
+/**
+ * Wait for the rules to change as a result of the thunk. The returned promise will resolve when the rules being served have changed.
+ * @param thunk Should cause the rules the RuleServer is serving to change some way.
+ */
+async function waitForRuleChange(thunk): Promise<void> {
+    const initialRules = await currentRules();
+    let rules = initialRules;
+    LogService.debug('PolicyConsumptionTest', `Rules before we wait for them to change: ${JSON.stringify(rules, null, 2)}`);
+    await thunk();
+    while (rules.since === initialRules.since) {
+        await new Promise<void>((resolve, reject) => {
+            setTimeout(() => resolve(), 500);
```

Comment:
Prefer the simpler:

```js
setTimeout(resolve, 500)
```

Other places, too.

Reply:
Ooops